spec/system/provider_interface/provider_manages_application_interviews_spec.rb

```diff
@@ -114,13 +118,28 @@ def and_an_interview_has_been_created(date)
 
   alias_method :and_another_interview_has_been_created, :and_an_interview_has_been_created
 
+  def when_the_first_interview_has_happened
+    Timecop.travel(30.minutes)
+    visit current_path
+  end
+
+  def then_i_see_the_upcoming_interview_under_the_correct_heading
+    expect(page).to have_css('.app-interviews > :nth-child(2)', text: 'Upcoming interviews')
+    expect(page).to have_css('.app-interviews > :nth-child(3)', text: '3 March 2020')
```

Comment:
Can we instead use the `data-qa` tag with a unique descriptive value, maybe `interview-#{interview.id}` to ensure that the spec can never fail as we've has a couple of issues with selectors that expect things to be in a certain order in order to work.

Reply:
This spec is trying to check that the page is in the correct order. I want it to fail if we don't get the heading followed by the interview. Let me know if I've missed your point though!

Reply:
We had some flaky specs using this type of selector which were caused due to additional data being generated. Probably because our factories tend to be extended without taking into consideration existing usage. I guess we can keep an eye on it for now.